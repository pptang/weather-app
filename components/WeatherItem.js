// @flow
import * as React from 'react';

import getWeatherIconClass from '../utils/weatherIconClassMap';

type Props = {
  code: string,
  day: string,
  high: string,
  low: string,
  isSelected: boolean,
};

const WeatherItem = (props: Props) => (
  <article className={`weatherItemWrapper ${props.isSelected ? 'isSelected' : ''}`}>
    <h2>{props.day}</h2>
    <figure>
      <i className={`wi ${getWeatherIconClass(props.code)}`} />
    </figure>
    <div>
      <span>{props.low}&#176;</span>~<span>{props.high}&#176;</span>
    </div>
    <style jsx>
      {`
        .weatherItemWrapper {
          flex: 1;
          display: flex;
          justify-content: center;
          align-items: center;
          flex-direction: column;
          height: 100%;
          background-color: #ffffff;
          color: #2e3849;
          cursor: pointer;
        }
        .weatherItemWrapper:hover {
          color: #4581cf;
        }
        .isSelected {
          color: #4581cf;
          font-size: 17px;
        }
      `}
    </style>
  </article>
);

export default WeatherItem;
