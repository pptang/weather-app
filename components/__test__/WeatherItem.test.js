import renderer from 'react-test-renderer';
import React from 'react';

import WeatherItem from '../WeatherItem';

describe('Pages With Snapshot Testing', () => {
  it('snapshot WeatherItem Component', () => {
    const testProps = {
      code: '25',
      day: 'Tue',
      high: 15,
      low: 10,
    };
    const component = renderer.create(<WeatherItem {...testProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
