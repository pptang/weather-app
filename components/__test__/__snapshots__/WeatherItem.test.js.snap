// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages With Snapshot Testing snapshot WeatherItem Component 1`] = `
<article
  className="jsx-3266424908 weatherItemWrapper "
>
  <h2
    className="jsx-3266424908"
  >
    Tue
  </h2>
  <figure
    className="jsx-3266424908"
  >
    <i
      className="jsx-3266424908 wi wi-cloudy"
    />
  </figure>
  <div
    className="jsx-3266424908"
  >
    <span
      className="jsx-3266424908"
    >
      10
      °
    </span>
    ~
    <span
      className="jsx-3266424908"
    >
      15
      °
    </span>
  </div>
</article>
`;
